crates/bevy_core/src/lib.rs

```diff
@@ -44,7 +44,9 @@ impl Plugin for CorePlugin {
             .register_type::<Labels>()
             .register_type::<Range<f32>>()
             .register_type::<Timer>()
-            .add_system_to_stage(CoreStage::First, time_system.system())
+            // time system is added as an "exclusive system" to ensure it runs before other systems in CoreStage::First
+            // this also ensures that it runs before other exclusive systems added after CorePlugin
+            .add_system_to_stage(CoreStage::First, time_system.exclusive_system())
```

Comment:
Ahh I thought it used "insertion order" as the "default" order for exclusive systems. Is there a reason _not_ to do that given that exclusive systems can't run in parallel? Given that _all_ exclusive systems are ambiguous with each other, that feels preferable to requiring labels on everything (or adding everything to ambiguity sets).

That won't change what we decide to do here. "Time" is core enough that a label makes sense.

Reply:
There's a bit of history here: exclusive systems just ended up being that way during #1144 (I really do not remember how by now), I raised whether we should keep that as one of the questions during final stages of the implementation, and per your own [wishes](https://github.com/bevyengine/bevy/pull/1144#issuecomment-766493862) they stayed this way 😄

Since then my rationale pretty much remained "remove all traces of implicit order to prevent outright any issues that might be caused by it, and maintain API parity between the two system kinds".

Reply:
Haha totally forgot about that. Sorry for pulling you in two different directions. In my (admittedly weak) defense, my opinions on "ambiguity noise" have developed a lot since then and I don't think I fully groked the implications of that design choice at the time.

We have created a situation where _every_ exclusive system must be labeled and order defined explicitly relative to other exclusive systems. Thats not a comfortable api imo (especially in a cross plugin context).

Reply:
> We have created a situation where every exclusive system must be labeled and order defined explicitly relative to other exclusive systems. Thats not a comfortable api imo (especially in a cross plugin context).

Agreed on both points, although I'm not sure what to do about it.

Reply:
I agree. I forgot to mention, there is a planned solution that builds on top of what we have now: the "simple case" APIs like `with_system(a).then(b).and(c)` which encodes the graph "a -> (b | c)" - meaning, "b" and "c" can run in parallel but both have to run after "a". Whether it makes sense to go about it this way for exclusive systems specifically I haven't thought about yet.